src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -108,7 +108,7 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
 
-        if (model.hasPerson(editedPerson)) {
+        if (model.hasName(editedPerson) && !editedPerson.getName().equals(personToEdit.getName())) {
```

Comment:
why need to check if model.hasName(editedPerson)?

Reply:
This just checks that they dont repeat the name again:
e.g. contact's name is current abc
editc 1 n/abc won't work because model has abc and editedperson.getname equals persontoedit.getname